src/node/interfaces.cpp

```diff
@@ -719,10 +719,16 @@ class ChainImpl : public Chain
     util::Result<void> checkChainLimits(const CTransactionRef& tx) override
     {
         if (!m_node.mempool) return {};
-        LockPoints lp;
-        CTxMemPoolEntry entry(TxGraph::Ref(), tx, 0, 0, 0, 0, false, 0, lp);
         LOCK(m_node.mempool->cs);
-        return m_node.mempool->CheckPackageLimits({tx}, entry.GetTxSize());
+        // Use CTxMemPool's ChangeSet interface to check whether the chain
+        // limits would be violated. Note that the changeset will be destroyed
+        // when it goes out of scope.
+        auto changeset = m_node.mempool->GetChangeSet();
```

Comment:
In "Check cluster limits when using -walletrejectlongchains"  1102ac7f74ac2f48760b46be58c7deb70fa727cf

nit: also state that it should be safe to create a new change set here because we lock mempool cs.

Reply:
Seems redundant, given that we have an `EXCLUSIVE_LOCKS_REQUIRED` annotation on `GetChangeSet()` already? Will leave this as-is.